pkg/kapp/preflight/registry.go

```diff
@@ -51,33 +52,32 @@ func (c *Registry) Type() string {
 }
 
 // Set takes in a string in the format of
-// CheckName={true||false},...
+// CheckName,...
 // and sets the specified preflight check
-// as enabled if true, disabled if false
+// as enabled if listed, otherwise, sets as
+// disabled if not listed.
 // Returns an error if there is a problem
 // parsing the preflight checks
 func (c *Registry) Set(s string) error {
 	if c.known == nil {
 		return nil
 	}
 
+	enabled := map[string]struct{}{}
```

Comment:
Non blocking, but using the [sets package](https://pkg.go.dev/k8s.io/apimachinery/pkg/util/sets) would make it so this can be a new `sets.Set` instead of doing this `map[string]struct{}{}` setup. Then you can do things like `sets.Insert(enabled, "key")` and `enabled.Has("key")`

Reply:
In this case it was a temporary storage, avoid disabling everything, then enabling only things we cared about. We don't actually care about the the value, just which one's have been enabled.
I suppose, if a `Set` were used, we could set the Set to disabled, then run through the passed in commands, and go through the Set and enable/disable...

There are so many ways to do this!

Reply:
A `sets.Set` is used exactly for the purpose of only caring about a key without the value. It's a generic map to an empty struct:
>type Set[T [comparable](https://pkg.go.dev/builtin#comparable)] map[T][Empty](https://pkg.go.dev/k8s.io/apimachinery/pkg/util/sets#Empty)

Again, non-blocking because like you said there are many ways to do this but this is a nice helper for this that is in the kube apimachinery package :)

Reply:
**[thought]** Would have been our first use of generics 😆 , but nothing to add on here. Sounds good to me either way!